### PR TITLE
fix(discover-quick-context): Added fixes to minor bugs.

### PR DESCRIPTION
**Pull request by Abdkhan14**
Fixed:
1. Cursor shouldn't be pointer on the icon.
2. Clicking hovercard shouldn't lead to issue details.
3. OverflowEllipsis wasn't working for shortid links (non-qcid fix)
4. 404 on clicking link to go to issue details page, caused by groupstore conflict.

### CODE REVIEW

File: static/app/views/eventsV2/table/quickContext.tsx [hunk 1]
```diff
@@ -168,6 +168,14 @@ export function QuickContextHoverWrapper(props: ContextProps) {
   const [data, setData] = useState<Group | null>(null);
   const delayOpenTimeoutRef = useRef<number | undefined>(undefined);
 
+  useEffect(() => {
+    return () => {
+      // Prevents conflicts in GroupStore usage as we navigate to issue details page
+      // using the shortId link.
+      GroupStore.reset();
+    };
+  }, []);
```
Comment (narsaynorath) [inline static/app/views/eventsV2/table/quickContext.tsx:177]: I'm still not entirely sure if there's a nicer way to handle this. Could we split this off into a separate PR to follow up on?

**Review by narsaynorath — COMMENTED** [review]
Just left some comments about splitting off the GroupStore thing, and a suggestion around the FieldShortId, I'd rather not introduce `OverflowFieldShortId` if we already have components that can do what we want. My suggestion did come with caveats but I wanted to offer it in case we prefer that.

Re-request me when you want me to look again!

File: static/app/views/eventsV2/table/quickContext.tsx [hunk 2]
```diff
@@ -250,6 +258,7 @@ const StyledHovercard = styled(Hovercard)`
 
 const StyledIconInfo = styled(IconInfo)<{ishovering: number}>`
   color: ${p => (p.ishovering ? p.theme.gray300 : p.theme.gray200)};
+  cursor: default;
```
[the remark this answers is not in the document]
Comment (Abdkhan14) [reply static/app/views/eventsV2/table/quickContext.tsx:261]: Agree.

File: static/app/views/eventsV2/table/quickContext.tsx (hunk 1 above)
Comment (Abdkhan14) [reply static/app/views/eventsV2/table/quickContext.tsx]: Done!

**Review by narsaynorath — APPROVED** [review]

**Pull request merged**